### PR TITLE
Testflows tests for Extended Precision Data Types: Int128, UInt128, Int256, UInt256, Decimal256

### PR DESCRIPTION
**Pull request by MyroTk**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Testflows tests for the big integers to make sure supported functions work and not supported functions do not cause crashes.

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Testing for big ints using the following functions:
* Arithmetic
* Array, tuple, and map
* Bit
* Comparison
* Conversion
* Logical
* Mathematical
* Null
* Rounding
- Creating a table with columns that use the data types

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/testflows/extended_precision_data_types/configs/clickhouse/config.d/ssl.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<yandex>
```
Comment (alexey-milovidov) [inline tests/testflows/extended_precision_data_types/configs/clickhouse/config.d/ssl.xml:1]: Garbage?

Comment (MyroTk) [reply tests/testflows/extended_precision_data_types/configs/clickhouse/config.d/ssl.xml:1]: Yes. Will remove the unnecessary SSL files.

**Comment by alexey-milovidov** [comment]
Also please remove *all* unnecessary configs.

File: tests/testflows/extended_precision_data_types/configs/clickhouse/config.d/ports.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0"?>
```
Comment (alexey-milovidov) [inline tests/testflows/extended_precision_data_types/configs/clickhouse/config.d/ports.xml:1]: This is unneeded.

File: tests/testflows/extended_precision_data_types/requirements/requirements.md [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+
+## Terminology
+
+### Extended Precision Data Types
+
+* Int128 - [-170141183460469231731687303715884105728 : 170141183460469231731687303715884105727]
+* UInt128 - [0 : 340282366920938463463374607431768211455]
+* Int256 - [-57896044618658097711785492504343953926634992332820282019728792003956564819968 : 57896044618658097711785492504343953926634992332820282019728792003956564819967]
+* UInt256 - [0 : 115792089237316195423570985008687907853269984665640564039457584007913129639935]
+* Decimal256 - [ -1 * 10^(76 - S): 1 * 10^(76 - S) ], where S is the precision.
```
Comment (alexey-milovidov) [inline tests/testflows/extended_precision_data_types/requirements/requirements.md:75]: Does it include the bounds?

**Comment by vzakaznikov** [comment]
@MyroTk, tests failed because of

**Comment by vzakaznikov** [comment]
@alexey-milovidov, we see these `Exit 70` errors randomly when we try to bring up clickhouse containers. Any ideas?
For example,
```
               723ms                    [bash] ---------------------------------------------------------------------------------------------------------
-----------------------------
               723ms                    [bash] extended-precision-data-type_env_clickhouse1_1   bash -c clickhouse server  ...   Exit 70
               724ms                    [bash] extended-precision-data-type_env_clickhouse2_1   bash -c clickhouse server  ...   Exit 70
               724ms                    [bash] extended-precision-data-type_env_clickhouse3_1   bash -c clickhouse server  ...   Exit 70
               724ms                    [bash] extended-precision-data-type_env_mysql1_1        docker-entrypoint.sh mysqld      Up (health: starting)  
 3306/tcp, 33060/tcp
               724ms                    [bash] extended-precision-data-type_env_zookeeper_1     /docker-entrypoint.sh zkSe ...   Up (healthy)  
```
then on the next re-try we see
```
               564ms                    [bash] ---------------------------------------------------------------------------------------------------------
-----------------------------
               566ms                    [bash] extended-precision-data-type_env_clickhouse1_1   bash -c clickhouse server  ...   Exit 70
               570ms                    [bash] extended-precision-data-type_env_clickhouse2_1   bash -c clickhouse server  ...   Up (health: starting)  
 8123/tcp, 9000/tcp, 9009/tcp
               574ms                    [bash] extended-precision-data-type_env_clickhouse3_1   bash -c clickhouse server  ...   Exit 70
               576ms                    [bash] extended-precision-data-type_env_mysql1_1        docker-entrypoint.sh mysqld      Up (health: starting)  
 3306/tcp, 33060/tcp
               579ms                    [bash] extended-precision-data-type_env_zookeeper_1     /docker-entrypoint.sh zkSe ...   Up (healthy)     
```
So sometimes clickhouse containers are up and sometimes they exit with error code 70.

**Comment by vzakaznikov** [comment]
@MyroTk, please fix style check.

```
/ClickHouse/tests/testflows/extended_precision_data_types/extended-precision-data-type_env/docker-compose.yml
  68:35     error    no new line character at the end of file  (new-line-at-end-of-file)

2021-05-20 17:43:31,478 Result parsed
2021-05-20 17:43:31,479 Result written
```

**Comment by alexey-milovidov** [comment]
I have no idea what error 70 means in Docker. Check the stderr.

File: tests/testflows/extended_precision_data_types/requirements/requirements.md (hunk 3 above)
Comment (MyroTk) [reply tests/testflows/extended_precision_data_types/requirements/requirements.md:75]: Yes

File: tests/testflows/extended_precision_data_types/requirements/requirements.md [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+
+## Terminology
+
+### Extended Precision Data Types
+
+* Int128 - [-170141183460469231731687303715884105728 : 170141183460469231731687303715884105727]
+* UInt128 - [0 : 340282366920938463463374607431768211455]
+* Int256 - [-57896044618658097711785492504343953926634992332820282019728792003956564819968 : 57896044618658097711785492504343953926634992332820282019728792003956564819967]
+* UInt256 - [0 : 115792089237316195423570985008687907853269984665640564039457584007913129639935]
+* Decimal256 - [10^(76 - S)-1: 10^(76 - S)+1], where S is the scale.
```
Comment (alexey-milovidov) [inline tests/testflows/extended_precision_data_types/requirements/requirements.md:75]: Does not look correct, see

`SELECT '1000000000'::Decimal32(0)`

Comment (MyroTk) [reply tests/testflows/extended_precision_data_types/requirements/requirements.md:75]: This is only for Decimal256. Decimal32 range would be ( -1 * 10^(9 - S), 1 * 10^(9 - S) )

**Comment by MyroTk** [comment]
@alexey-milovidov The exp function when run in the PR returned '2.7182818284606256', however running locally on the same ClickHouse version, exp returns '2.718281828460626'. I used the deb package from the PR to update my local build. What could be causing this inconsistency?

**Comment by alexey-milovidov** [comment]
@MyroTk ClickHouse is using `libc` from the system. Different versions of `libc` may have different behaviour.
We will work on static linking with `libc` to avoid this issue.

**Comment by MyroTk** [comment]
@alexey-milovidov The testflows check is green, should be ready for merge.

**Comment by alexey-milovidov** [comment]
PS. @MyroTk Decimal data type does not include the bounding powers of ten. It's from -...999.999... to ...999.999...

**Comment by MyroTk** [comment]
> 
> 
> PS. @MyroTk Decimal data type does not include the bounding powers of ten. It's from -...999.999... to ...999.999...

I specified in the requirements that the boundaries for Decimal are exclusive.

**Pull request merged**